### PR TITLE
Fixes Object keyword not being considered to be an escaped type name.

### LINKED ISSUE #41: Parser seems unable to handle type _Object correctly
(Originally from https://github.com/eProsima/Fast-DDS-Gen/issues/44.)

I'm trying to use fastrtpsgen 1.0.4 with the option -typeros2 and a struct called _Object_. This seems to be a reserved keyword in IDL, so I tried to escape the name by putting _ in front. While the conversion of the struct seems to work correctly, the type cannot be used in another struct defined in the IDL.

See the following minimal example:

    module pk_msgs {
            module msg {
                    struct _Object {
                            int64 id;
                    };

                    typedef sequence<_Object> ObjectSeq;

                    struct Objects {
                            ObjectSeq objs;
                    };
            };
    };

This produces the following error when trying to generate code for it with fastrtpsgen:

    openjdk version "1.8.0_265"
    OpenJDK Runtime Environment (build 1.8.0_265-8u265-b01-0ubuntu2~20.04-b01)
    OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)
            Loading templates...
            Processing the file obj.idl...
            obj.idl:7:19: error: '_Object' was not defined previously
            obj.idl:10:3: error: 'ObjectSeq' was not defined previously
    Exception in thread "main" java.lang.NullPointerException
            at com.eprosima.fastrtps.fastrtpsgen.execute(fastrtpsgen.java:343)
            at com.eprosima.fastrtps.fastrtpsgen.main(fastrtpsgen.java:1207)

### PR DESCRIPTION
**Pull request by LuisGP**
Fixes #41

### CODE REVIEW

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4 [hunk 1]
```diff
@@ -1113,12 +1113,24 @@ any_type returns [TypeCode typecode]
...
+        // Find typecode in the global map.
+        $typecode = ctx.getTypeCode("Object");
+
+        if($typecode == null)
+        {
+            throw new ParseException(tk, ". Object type is not supported");
+        }
+        else
+        {
+            System.out.println("Encontrado un Object...");
```
Comment (jwillemsen) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4:1131]: English?

Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4:1131]: Oops :)

Comment (MiguelCompany) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4]: I was going to say the same!

**Comment by jwillemsen** [comment]
Maybe add a test case as reproducer here?

**Comment by MiguelCompany** [comment]
> Maybe add a test case as reproducer here?

It would be nice to have them here, but I think it is enough to add them to Fast-DDS-Gen

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**